### PR TITLE
Fix conda recipe

### PR DESCRIPTION
**Pull request by groutr**
Synced changes from conda-recipes.

### CODE REVIEW

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -11,11 +11,7 @@ build:
   # so we have to exempt them from the list of initial files
   # for conda-build to realize they should be included.
   always_include_files:
-    - bin/activate
-    - bin/deactivate
-    - bin/conda
-  entry_points:
-    - conda = conda.cli.main:main
+    - bin/conda  [unix]
```
Comment (kalefranz) [inline conda.recipe/meta.yaml]: @groutr why not use `entry_points`? Against @ilanschnell's preference, I refactored `setup.py` to be all-in on setuptools.  So, `entry_points` there, with `conda build` overwriting the crappy `bin/` code setuptools creates.  We can also now `rm -rf bin/conda` in the repo I think.

File: conda.recipe/meta.yaml [hunk 2]
```diff
@@ -11,11 +11,7 @@ build:
   # so we have to exempt them from the list of initial files
   # for conda-build to realize they should be included.
   always_include_files:
-    - bin/activate
-    - bin/deactivate
-    - bin/conda
```
Comment (ilanschnell) [inline conda.recipe/meta.yaml]: @groutr Don't remove these.  In fact there should be a selector for `unix`, and then other files for `win`

Comment (groutr) [inline conda.recipe/meta.yaml]: I thought that bin/activate and bin/deactivate are provided by conda-env?

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (groutr) [inline conda.recipe/meta.yaml]: Oh nice.  I didn't realize that.

File: conda.recipe/meta.yaml (hunk 2 above)
Comment (msarahan) [inline conda.recipe/meta.yaml]: They are for now.  #1727 changes that, though.

File: conda.recipe/meta.yaml (hunk 1 above)
Comment (groutr) [inline conda.recipe/meta.yaml]: Adding back entry points leads to this traceback:
```
+ conda --version
Traceback (most recent call last):
  File "/Users/rgrout/miniconda/envs/_test/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main.main())
AttributeError: 'function' object has no attribute 'main'
```

Comment (groutr) [inline conda.recipe/meta.yaml]: Just realized that it should be `conda.cli:main`

**Comment by groutr** [comment]
Looks like tests need to be updated with the removal of bin/conda

**Comment by kalefranz** [comment]
GOTO #2135

**Pull request closed without merging**